crates/sui-graphql-rpc/src/server/watermark_task.rs

```diff
@@ -96,10 +111,42 @@ impl WatermarkTask {
         self.watermark.clone()
     }
 
+    pub(crate) fn chain_id_lock(&self) -> ChainIdentifierLock {
+        self.chain_identifier.clone()
+    }
+
     /// Receiver for subscribing to epoch changes.
     pub(crate) fn epoch_receiver(&self) -> watch::Receiver<u64> {
         self.receiver.clone()
     }
+
+    // Fetch the chain identifier (once) from the database and cache it.
+    async fn get_and_cache_chain_identifier(&self, interval: &mut Interval) -> bool {
+        loop {
+            tokio::select! {
+                _ = self.cancel.cancelled() => {
+                    info!("Shutdown signal received, terminating attempt to get chain identifier");
+                    return false;
+                },
+                _ = interval.tick() => {
+                    // we only set the chain_identifier once.
+                    let chain = match ChainIdentifier::query(&self.db).await  {
+                        Ok(Some(chain)) => chain,
+                        Ok(None) => continue,
+                        Err(e) => {
+                            error!("{}", e);
```

Comment:
nit: some extra context for the error

```suggestion
                            error!("Failed to fetch chain identifier: {}", e);
```